pipeline_dp/histograms.py

```diff
@@ -99,6 +100,46 @@ def quantiles(self, q: List[float]) -> List[int]:
         return result[::-1]
 
 
+def compute_ratio_dropped(
+        contribution_histogram: Histogram) -> Sequence[Tuple[int, float]]:
+    """Computes ratio of dropped data for different bounding thresholds.
+
+    For each histogram bin.lower it computes what would the ratio of data
+    dropped because of contribution bounding when it is taken as bounding
+    threshold (e.g. in case of L0 histogram bounding_threshold is
+    max_partition_contribution).
+
+    Args:
+        contribution_histogram: histogram of contributions. It can be L0, L1,
+          Linf contribution bounding histogram.
+
+    Returns:
+        A sequence of sorted pairs
+        (bounding_threshold:int, ratio_data_draped:float). Where
+        bounding_thresholds are all lower of histogram.bins and
+        histogram.max_value.
+    """
+    if not contribution_histogram.bins:
+        return []
+    dropped = elements_larger = 0
+    total_sum = contribution_histogram.total_sum()
+    ratio_dropped = []
+    bins = contribution_histogram.bins
+    previous_value = bins[-1].lower  # lower of the largest bin.
+    if contribution_histogram.max_value != bins[-1].lower:
+        # Add ratio for max_value when max_value is not lower in bins.
```

Comment:
I am not sure I understand this. Why do we need this step?

Reply:
It's mostly convenience thing. Those ratios will be used for finding candidates for max_partition_contributed (and other bounds), and it's convenient to have max contribution as one of the candidates